python_modules/dagster/dagster/_core/definitions/auto_materialize_rule_impls.py

```diff
@@ -281,17 +281,22 @@ def passes(
                         AssetKeyPartitionKey(asset_key, record.partition_key)
                     )
 
+        run_ids_with_required_tags = set()
+
         if len(asset_partitions_by_latest_run_id) > 0:
-            run_ids_with_required_tags = (
-                context.legacy_context.instance_queryer.instance.get_run_ids(
-                    filters=RunsFilter(
-                        run_ids=list(asset_partitions_by_latest_run_id.keys()),
-                        tags=self.latest_run_required_tags,
-                    )
+            run_step = int(os.getenv("DAGSTER_ASSET_DAEMON_RUN_TAGS_RUN_FETCH_LIMIT", "1000"))
```

Comment:
should we make this so if its set to `0` or something it does the previous impl just for extra easy of behavior tweak ?